src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -719,6 +719,81 @@ def iot_hub_manual_failover(cmd, client, hub_name, failover_region, resource_gro
+
+
+def pnp_update_repository(cmd, client, repo_endpoint, repo_id, repo_name):
+    return _pnp_create_update_repository(cmd, client, repo_endpoint, repo_name, repo_id)
+
+
+def pnp_list_repository(cmd, client, repo_endpoint):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).get_repositories_async(api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_get_repository(cmd, client, repo_endpoint, repo_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).get_repository_async(repo_id, api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_remove_repository(cmd, client, repo_endpoint, repo_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).delete_repository_async(repo_id, api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_track_provision_status(cmd, client, repo_endpoint, repo_id, track_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).get_provision_status(repo_id, track_id, api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_create_authkeys(cmd, client, repo_endpoint, repo_id, user_role):
+    return _pnp_create_update_authkeys(cmd, client, repo_endpoint, repo_id, user_role)
+
+
+def pnp_update_authkeys(cmd, client, repo_endpoint, repo_id, key_id, user_role):
+    return _pnp_create_update_authkeys(cmd, client, repo_endpoint, repo_id, user_role, key_id)
+
+
+def pnp_list_authkeys(cmd, client, repo_endpoint, repo_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).get_keys_async(repository_id=repo_id, api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_get_authkeys(cmd, client, repo_endpoint, repo_id, key_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).get_key_async(repo_id, key_id, api_version=client.api_version, custom_headers=headers)
+
+
+def pnp_remove_authkeys(cmd, client, repo_endpoint, repo_id, key_id):
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    return _get_pnp_client(repo_endpoint).delete_key_async(key_id, repo_id, api_version=client.api_version, custom_headers=headers)
+
+
+def _pnp_create_update_repository(cmd, client, repo_endpoint, repo_name, repo_id=None):
+    from .digitaltwinrepositoryprovisioningservice.models import RepositoryUpsertRequestProperties
+    headers = {'Authorization': '{}'.format(_generate_bearer_token(cmd))}
+    repositoryUpsertRequestProperties = RepositoryUpsertRequestProperties(id=repo_id, name=repo_name)
+    return _get_pnp_client(repo_endpoint).create_or_update_repository_async(api_version=client.api_version,
```

Comment:
is there a synchronous version of this create_or_update method. 

If so, then you can leverage [--no-wait ](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#standard-command-types)to decide which method you would run, depending on user input (see the section on the wait command).

Reply:
no, the only async version is there now, maybe in future then I'll update